Makefile

```diff
@@ -6,15 +6,27 @@ default: fmtcheck vet build
 # test runs the test suite and vets the code
 test: fmtcheck
 	@echo "==> Running Tests"
-	cd govcd && go test -timeout=45m -check.vv .
+	cd govcd && go test -tags "functional" -timeout=45m -check.vv .
```

Comment:
`make test` should run all tests, not only functional.

Reply:
> `make test` should run all tests, not only functional.

This is intentional. For now, we only have tests that use the gocheck library. When we start using a different framework (of which we have already some samples in our personal repos), we don't want to mix them, because both frameworks have some "magic" that starts the tests, and we want to avoid possible conflicts.
The purpose of `make test` is to run the functional tests without `TF_ACC`, just to see if the tests runs.
If we want to check that the tests compile, running `make` will do it, as it will compile using the `ALL` tag

Reply:
What we need is an easy and safe way to run all the tests. How to do that?

Reply:
How about we get to that point when we have non `functional` tests?
By then, we will know what kind of tests we want to implement, how to enable them, and whether they can run together or separately. 
This framework has paved the way to add more tests safely.
If we design the next tests cleanly, we can run them with two separate commands, possibly in parallel.